### PR TITLE
[TT-10881] Make BaseMiddleware a pointer

### PR DESCRIPTION
**Pull request by titpetric**
This resolves vet warnings about copying a mutex value.

Before: 189 warnings,
After: 7 warnings

```
# go vet -json ./... 2>&1 | summary vet
Warnings by message:
3 loop variable tc captured by func literal
2 func passes lock by value: github.com/TykTechnologies/tyk/gateway.APISpec
1 call of tc.reloadAPI copies lock value: github.com/TykTechnologies/tyk/gateway.APISpec
1 method Seek(offset int64, whence int64) (int64, error) should have signature Seek(int64, int) (int64, error)

Warnings by filename:
3 /root/tyk/tyk/gateway/handler_success_test.go
2 /root/tyk/tyk/apidef/oas/server_test.go
1 /root/tyk/tyk/test/util_test.go
1 /root/tyk/tyk/gateway/reverse_proxy.go

Total: 7
```

https://tyktech.atlassian.net/browse/TT-10881

### CODE REVIEW

**Comment by buger** [comment]
**API tests result - postgres15-murmur64 env: skipped** :no_entry_sign: 
Branch used: refs/pull/5914/merge
Commit:  
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7348838821)

**Comment by buger** [comment]
**API tests result - mongo44-sha256 env: skipped** :no_entry_sign: 
Branch used: refs/pull/5914/merge
Commit:  
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7348838821)

**Comment by buger** [comment]
**API tests result - postgres15-sha256 env: skipped** :no_entry_sign: 
Branch used: refs/pull/5914/merge
Commit:  
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7348838821)

**Comment by buger** [comment]
**API tests result - mongo44-murmur64 env: skipped** :no_entry_sign: 
Branch used: refs/pull/5914/merge
Commit:  
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/7348838821)

File: gateway/middleware.go [hunk 1]
```diff
@@ -46,6 +46,7 @@ var (
 type TykMiddleware interface {
 	Init()
 	Base() *BaseMiddleware
+
```
Comment (jeffy-mathew) [inline gateway/middleware.go:49]: unnecessary line break?

**Review by jeffy-mathew — APPROVED** [review]

**Pull request merged**